### ISSUE TITLE
AssertionError

### ISSUE THREAD

**Issue by inconnu11** [kan-bayashi/ParallelWaveGAN#233]
Hi, I modify the sample rate, fft_size and hop size as 16000, 1024 and 256 of the _parallel_wavegan_v1.yaml_ of the _VCTK_ corpus. I also modify the upsample_scales as [4, 4, 4, 4] to ensure the prodcut is the same as hop size. And the batch_max_steps is set as 16384 to ensure can be divided by hop size.
But when training, it still says **AssertionError** which occurs in the `assert len(x) == len(c) * self.hop_size`.
When I replace the `assert` with judging whether the quality is true by `if xx != yy` it doesn't print any prompt as I code in the `if` statement.
How can I fix this?

**Comment by kan-bayashi** [comment]
Did you re-run stage 1?

**Comment by inconnu11** [comment]
Yes. I set the start stage as -1 and stop stage as 100 which means it will re-process the data every time I run starting from the run.sh.

**Comment by kan-bayashi** [comment]
I’m not sure what is happened.
Could you paste your config and the head of wav.scp?

**Comment by inconnu11** [comment]
my config is:
allow_cache: true
batch_max_steps: 16384
batch_size: 6
config: conf/mysetting.yaml
dev_dumpdir: dump/dev_all/norm
dev_feats_scp: null
dev_segments: null
dev_wav_scp: null
discriminator_grad_norm: 1
discriminator_optimizer_params:
  eps: 1.0e-06
  lr: 5.0e-05
  weight_decay: 0.0
discriminator_params:
  bias: true
  conv_channels: 64
  in_channels: 1
  kernel_size: 3
  layers: 10
  nonlinear_activation: LeakyReLU
  nonlinear_activation_params:
    negative_slope: 0.2
  out_channels: 1
  use_weight_norm: true
discriminator_scheduler_params:
  gamma: 0.5
  step_size: 200000
discriminator_train_start_steps: 100000
distributed: false
eval_interval_steps: 1000
fft_size: 1024
fmax: 7600
fmin: 125
format: hdf5
generator_grad_norm: 10
generator_optimizer_params:
  eps: 1.0e-06
  lr: 0.0001
  weight_decay: 0.0
generator_params:
  aux_channels: 80
  aux_context_window: 2
  dropout: 0.0
  gate_channels: 128
  in_channels: 1
  kernel_size: 3
  layers: 30
  out_channels: 1
  residual_channels: 64
  skip_channels: 64
  stacks: 3
  upsample_net: ConvInUpsampleNetwork
  upsample_params:
    upsample_scales:
    - 4
    - 4
    - 4
    - 4
  use_weight_norm: true
generator_scheduler_params:
  gamma: 0.5
  step_size: 200000
global_gain_scale: 1.0
hop_size: 256
lambda_adv: 4.0
log_interval_steps: 100
num_mels: 80
num_save_intermediate_results: 4
num_workers: 2
outdir: exp/train_nodev_all_vctk_mysetting
pin_memory: true
pretrain: ''
rank: 0
remove_short_samples: true
resume: ''
sampling_rate: 16000
save_interval_steps: 5000
stft_loss_params:
  fft_sizes:
  - 1024
  - 2048
  - 512
  hop_sizes:
  - 120
  - 240
  - 50
  win_lengths:
  - 600
  - 1200
  - 240
  window: hann_window
train_dumpdir: dump/train_nodev_all/norm
train_feats_scp: null
train_max_steps: 1000000
train_segments: null
train_wav_scp: null
trim_frame_size: 1024
trim_hop_size: 256
trim_silence: true
trim_threshold_in_db: 20
verbose: 1
version: 0.4.8
win_length: null
window: hann

the wav.scp:
<img width="646" alt="image" src="https://user-images.githubusercontent.com/38338826/98921554-6a687d00-250c-11eb-9519-f9e1754327b4.png">

**Comment by inconnu11** [comment]
What's more, the Mel spectrogram is normalized to [-3, 3]? Where can I modify parameters to change the normalized scope to [-1, 1]?

**Comment by inconnu11** [comment]
The original  input wavs are 48k. What I need is 16k. And I think there is no need to down-sample the wavs manually.  Will the sox down-sample the wavs?

**Comment by kan-bayashi** [comment]
It looks no problem. I have no idea why the assertion error is happened.
If you have a still error, could you delete `data` and `dump` and try from scratch to reproduce the error?

> Will the sox down-sample the waves?

Right. Your wav.scp looks fine.

> What's more, the Mel spectrogram is normalized to [-3, 3]? Where can I modify parameters to change the normalized scope to [-1, 1]?

I used mean-var normalization to be mean 0 and variance 1.
Therefore, there is no explicit range.
IF you want to do it, change the scaler by yourself.
https://github.com/kan-bayashi/ParallelWaveGAN/blob/53d14969089b3d3229fe6bfce221234f25a9d836/parallel_wavegan/bin/compute_statistics.py#L91-L92
https://github.com/kan-bayashi/ParallelWaveGAN/blob/53d14969089b3d3229fe6bfce221234f25a9d836/parallel_wavegan/bin/normalize.py#L123-L134
